src/05-derivatives/04-functional-derivatives.md

```diff
@@ -47,51 +47,62 @@ The following metadata JSON fields are valid for derivative maps:
 | Threshold      | String describing threshold used for determining graph edges.             | `dcb`, `dcw`, `ecb`, `ecw` |
 | Method         | String describing method used to calculate measure.                       | `dcb`, `dcw`, `ecb`, `ecw` |
 
-## Time series and regressors
+## Time series
 
-For the purposes of this section, a time series or regressor is a numeric series
-of values corresponding to volumes in a BOLD series. Time series is specifically
-used to indicate an aggregate time series within an ROI, while regressor is used
-as a more general term indicating series that may be derived from other sources,
-such as head motion estimates or physiological recordings.
+A time series is an ordered series of numeric values in chronological order.
+For the purposes of this section, a "regressor" is a time series with one entry
+per volume in a BOLD series.
 
-Regressors will generally be stored as tables, with a row of column headers
-indicating the name of the regressor. In the case where every voxel has a
-regressor, then the data should be stored in a 4D NIfTI file.
+Time series will generally be stored as tables, with a row of column headers
+indicating the name of the series.
+In the case where every voxel has a time series, then the data should be stored
+in a 4D NIfTI file.
+
+All time series files MUST be accompanied by a data dictionary in JSON format,
+containing the following sampling information:
+
+| Field name | Definition |
```

Comment:
I'm not sure this needs to be explicitly described again here given that it's already covered in the core BIDS spec (see [here](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#tabular-files)). May be better to point readers to that, otherwise we risk introducing contradictions and minor differences in interpretation.

Also, is there a reason why we're making this mandatory? It's optional in the BIDS spec, so it seems inconsistent to mandate it in derivatives. I get that `SamplingFrequency` is required, but I'm not sure forcing descriptions of the columns is prudent.

Reply:
`SamplingFrequency` and `StartTime` are not defined in the BIDS-Raw that I can see, though `SamplingFrequency` has been added in the 1.2.0 electrophysiology sections. I was planning on opening an issue WRT _stim and _physio...

I think the logic of making descriptions required was that these are almost all going to be produced by tools, and without required metadata, the chances of misinterpretation are high. So it would be the responsibility of the tool to ensure that these exist. But I can't be sure about that, and if the consensus is that this is too high a burden, then we can change it to RECOMMENDED.

Reply:
They are sort of defined in text here: https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/06-physiological-and-other-continous-recordings.html. Changing this into a table would make it more clear.

Reply:
Ah, you're right. I was just Ctrl-F searching "StartTime".

Reply:
I think generating column names/descriptions is actually a fairly cumbersome thing in many cases, because someone has to write the content. I'm selfishly thinking here of tools like pliers, which can easily generate hundreds of columns, but don't necessarily provide sensible descriptions. I'm not sure auto-populating a JSON file with hundreds of entries like `"ClarifaiLabelExtractor_dog": "feature 'dog' extracted using ClarifaiLabelExtractor"` is an improvement.

That said, I'd be okay mandating column descriptions as long as we do the same for BIDS-Raw. The obvious problem with *that* though is that we would instantly invalidate pretty much all of the existing BIDS datasets, because relatively few people have bothered to provide data dictionaries. (I'm not sure if the latter observation is an argument in favor of mandating this or against it!)

Reply:
+1 on @chrisfilo's point of having this table in the "other-continuous-recordings" text, since from the description there it's not clear to me that the `SamplingFrequency` is REQUIRED while `StartTime` is OPTIONAL. Syncing these two and then pointing to that as @tyarkoni suggests seems sensible.

Regarding column descriptions: Is there a reason we have to mandate them in BIDS-Raw _now_, @tyarkoni ? Could we mandate them in derivatives and then in the core spec after 2.0 ? Or is the concern that this would likely create confusion ? I'm with @effigies in that I'd prefer to have them, if possible.

Reply:
I think this will be a source of ongoing confusion and extra work. Presumably the validator (once updated to handle derivatives) is going to have to fail derivatives that don't have a data dictionary, while passing raw datasets that don't. In the case of `_events.tsv` files, this will also almost certainly mean having to create sidecars, which are generally not provided in BIDS-raw datasets (because if you're not providing a data dictionary, there's no need for them). And for a lot of columns, descriptions are either unnecessary (because self-explanatory) or unavailable (as in the pliers example I gave above), so mandating them isn't an improvement.

Basically I think this risks losing sight of the original 80/20 rule... It's extra information that can make life better from a programmatic perspective, but can be cumbersome for users to add, and may deter fully BIDSifying one's data. In isolation, this one decisions is probably not a big deal, but every time we opt for the more comprehensive-but-effortful approach, we're raising the bar a little bit for new users/developers to achieve compliance, and I think the risk is that it starts to limit adoption at some point.

It's worth keeping in mind that we're probably the people least likely to appreciate the cost of incrementally adding requirements to the spec, because we're mostly already writing tools that meet the existing spec, so each additional change is not a big deal. But if you're new to BIDS, and you're contemplating writing a new tool that has to produce outputs that pass the validator, it's going to seem overwhelming at some point (if it doesn't already to some people) to stare down a giant document full of MANDATORY requirements.

Reply:
I think that's a really valuable perspective ! I also think the opposite perspective -- making sure the data retains maximum usability for users -- is equally important here, but that's a broader discussion than this one point :)

The concern I see (and it sounds like @effigies had) is not that this information makes life easier from a programmatic perspective, but that we end up seeing many NeuroStars questions saying something like "I don't know how to interpret this one field from tool X -- does anyone know what it means?"

If we think the frequency of those cases will be on the 20 side of the 80/20 (because of things like self-explanatory columns), then I'm ok to move to RECOMMENDED. But I don't imagine this decision will be because of the validator (since failing a derivatives dataset and passing a raw one seems feasible, at least to this naive developer), but because of these broader points on adoption vs (re)usability.

Reply:
Yeah, I think I'm good with downgrading to RECOMMENDED. It makes sense for tools that are trying to be maximally useful to use RECOMMENDED to identify targets of effort, but reducing the distance to conformant is definitely a good heuristic to keep in mind for the spec.

With that said, I seem to recall a push to a lot of mandatory metadata from the meeting. It may be worth checking through the main PR for how much is being demanded and potentially backing some more things down to RECOMMENDED status.